src/main/kotlin/com/google/coroutinestacks/ui/DAG.kt

```diff
@@ -0,0 +1,131 @@
+package com.google.coroutinestacks.ui
+
+import java.awt.Dimension
+import java.lang.Integer.max
+import java.util.*
+
+class DAG(
```

Comment:
Let's simplify the declaration `class DAG(traces: List<CoroutineTrace>)`.  And then we can compute map of children in the constructor. This way we will hide implementation details from the users of the class.